src/axiosStore.js

```diff
@@ -27,13 +32,14 @@ const axiosStore = (axiosInstance, store) => {
             } else {
               wrappedData = data;
             }
-            store.set(cacheKey, wrappedData);
-            return { ...wrappedData, __cacheKey: cacheKey };
+
+            return wrappedData;
           })
+          .then(normalizedData => callback(normalizedData))
+          .then(processedData => store.set(cacheKey, processedData))
+          .then(data => ({ ...data, __cacheKey: cacheKey, __fromCache: false }))
```

Comment:
@cvburgess I pulled this branch and used `npm link` to test these changes out a bit. Everything seems to be working as expected with most of our actions (including the new `callback` option) but I'm seeing some issues in actions where `__cacheKey` is set manually. Obviously manually setting `__cacheKey` is a pattern we want to deprecate, I just want to clarify if these are intentional breaking changes, or if these are unexpected. 


What I observed:
1. Actions that currently set their own `cacheKey` were not being cached as I would expect. I observed the same request for `/company` go out 4 times on the home page which is less than what's happening in QA at the moment.
1. Actions that currently set their own `cacheKey` can't utilize the `callback` option. This one is minor as this is a scenario that doesn't exist in the app today, just want to confirm this won't be supported.

Reply:
callback replaces the custom cacheKey FWIW

also, AFAICT this has always been broken, which is why i had to add the callback.

i think the way it works now is that custom cache keys jsut prevent us from overriding values in the store but it does not do what we thought it did. to reformat data and save it with the proper key, we need to leverage callbacks